src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -0,0 +1,483 @@
+     * @param generate Generate Baseline test files instead of running test
+     * @param pattern Pattern to match on the test description, matching tests will be run
+     * @throws IOException
+     * @throws SQLException
+     * @throws IllegalAccessException
+     * @throws InvocationTargetException
+     */
+    public void runTests(List<Tests> testLists, Connection conn, Boolean generate, String pattern)
+            throws IOException, SQLException, IllegalAccessException, InvocationTargetException {
+        Pattern p = Pattern.compile(pattern);
+        for (Tests tests : testLists) {
+            for (TestEntry testEntry : tests.tests) {
+                if (testEntry.skip_reason != null || !p.matcher(testEntry.description).matches()) {
+                    continue;
+                }
+                ResultSet rs = null;
+                // Run DatabaseMetadata function if it exists, takes precedence over query
+                if (testEntry.meta_function == null) {
+                    rs = executeQuery(testEntry, conn);
+                } else {
+                    rs = executeDBMetadataCommand(testEntry, conn.getMetaData());
+                }
+                assertNotEquals(rs, null);
+                if (generate) {
+                    TestGenerator.generateBaselineTestFiles(testEntry.description, rs);
+                } else {
+                    if (testEntry.ordered != null && testEntry.ordered) {
+                        validateResultsOrdered(testEntry, rs);
+                    } else {
+                        validateResultsUnordered(testEntry, rs);
+                    }
+                    validateResultSetMetadata(testEntry, rs.getMetaData());
+                }
+            }
+        }
+    }
+
+    private ResultSet executeQuery(TestEntry entry, Connection conn) throws SQLException {
+        Statement stmt = conn.createStatement();
+        return stmt.executeQuery(entry.sql);
+    }
+
+    private ResultSet executeDBMetadataCommand(TestEntry entry, DatabaseMetaData databaseMetaData)
+            throws SQLException, InvocationTargetException, IllegalAccessException {
+
+        List<Object> metadataFunction = entry.meta_function;
+        assertTrue(metadataFunction != null && metadataFunction.size() > 0);
+        String functionName = (String) metadataFunction.remove(0);
+        switch (functionName) {
+            case "getIndexInfo":
+                return executeGetIndexInfo(metadataFunction, databaseMetaData);
+            case "getBestRowIdentifier":
+                return executeGetBestRowIdentifier(metadataFunction, databaseMetaData);
+            case "getTables":
+                return executeGetTables(metadataFunction, databaseMetaData);
+            case "getUDTs":
+                return executeGetUDTs(metadataFunction, databaseMetaData);
+            default:
+                return executeDBMetadataHelper(functionName, metadataFunction, databaseMetaData);
```

Comment:
Thanks for the suggestion.  
I made a change that handles the different parameter types besides String.  Handled in the same function instead of running different functions that were based on the parameter types.  I tested that the methods with non-string parameters are working.